### ISSUE TITLE
Specifying the domain name of the calling page in the "redirect to" URL

### ISSUE THREAD

**Issue by Gitoffthelawn** [einaregilsson/Redirector#168]
Hi Einar!  I hope you are well.

Sometimes it would be very useful to be able to insert the domain name of the calling page into the "redirect to" URL.

An example will help:
On https://www.peazip.org/search-software-resources.html there is an input field to enter a search string.  When a string is entered, the site redirects to Google (yuck).  I want to use Redirector in these situations to redirect to a preferred search engine.

Because the resultant URL does not include "www.peazip.org", a variable (similar to $1) is needed to add the page's domain to the "redirect to" URL so that the request can be processed by a preferred search engine.

Is this currently possible?  If not, can it be a feature request?

**Comment by einaregilsson** [comment]
This is not possible right now. Adding it would require another field with another pattern. Sorry, I don't think it's something that would be used a lot and it would make the UI even more complicated.

**Comment by Gitoffthelawn** [comment]
Thanks for letting me know it's not possible right now.  I didn't think it was, but I wasn't 100% sure.

I don't think it would require another field or any UI changes, though.

It could just employ one or more replacement values like "$domain" or "$host", that would get replaced by the calling domain or host.

**Comment by einaregilsson** [comment]
Yeah, but we don't really have that information at the point we're doing the redirecting. In some cases we might be able to get it through the Referer header, but that's not always available now in ssl site and people can also set polices on whether it should be available or not.

**Comment by Gitoffthelawn** [comment]
Ah, good point.  I forgot that Redirector doesn't have any context information.

Just to refresh my memory, if you're on page "www.test1.com/page.html" and you click a link for "www.test2.com/page.html", Redirector has no idea that you're currently on "www.test1.com/page.html".  Do I have it correct?

**Comment by einaregilsson** [comment]
Yes, we just see an http request being made. Although, the originating site may be found in the Referer header, but as far as I know that may be omitted on https sites, and can also be omitted if the site puts int a `rel="noreferer"` attribute on the link, so it's not something we can really depend on.

I have thought about maybe going back to monitoring the actual address bar though, for two reasons. One, I think Chrome is going to remove the webrequest api we're using, and two, that would also catch all the sites like Twitter, Facebook etc that don't really load a new page but just load in new content and then change the url using the history api, Redirector currently can't redirect those, but would be able to if we just used the address bar. Of course, that wouldn't work for images, stylesheets etc. etc., so we'd either have to remove those options, or manage both which would be a big headache...

**Comment by Gitoffthelawn** [comment]
Hmmm... interesting conundrum.  I don't use Twitter or Facebook, so I wasn't aware of the games they are playing.  Are they using javascript to access the browser's history api to push a new url to the top of the browser's history stack?  If so, are there any restrictions to what they can push?

If you went back to the actual address bar, would performance change at all?

**Comment by einaregilsson** [comment]
They can push any url on their own site, yes. 

I don't think performance would suffer in a noticeable way, there are hooks that let you know before navigation starts, so I think I could stop it from ever sending the original http request, but I'm not 100% sure, I'd need to dig into it.

Right now I'm trying to update the html and remove angular.js from the code, then maybe I'll add a couple of the most requested features and do a release.
